rowboat/util/input.py

```diff
@@ -15,6 +15,9 @@ def parse_duration(raw, source=None, negative=False):
     if not raw:
         raise CommandError('Invalid duration')
 
+    if source == None:
```

Comment:
Try calling parse_duration with timedelta() (so 0 seconds) as source argument, for some reason the function will still choose datetime.utcnow() instead of the provided source arg timedelta() for the return value.